build_runner/lib/src/watcher/graph_watcher.dart

```diff
@@ -55,11 +55,13 @@ class PackageGraphWatcher {
                 .watch()
                 .where(_nestedPathFilter(w.node))
                 .handleError((Object e, StackTrace s) {
-              _logger.severe(
-                  'Error from directory watcher for package:${w.node.name}\n\n'
-                  'If you see this consistently then it is recommended that '
-                  'you enable the polling file watcher with '
-                  '--use-polling-watcher.');
+              _logger
+                ..severe('Error in watcher for package: ${w.node.name}: $e:$s')
```

Comment:
We probably don't want separate logs here, and the stack trace and exception should be passed using the explicit parameters for those

Reply:
This is the only change I actually want to keep:) the exception propagates to the shelf handler, and the stack trace gets shortened intentionally sp not much information is left. It looks like:

``
SocketException: Write failed (OS Error: Connection reset by peer, errno = 104), address = localhost, port = 35146
package:shelf/src/middleware/logger.dart  logRequests.<fn>.<fn>.<fn>
``

Or it would have a one-liner stack pointing to the shelf handler if I disable logging requests.

Reply:
Update - sorry, my mistake, this is just my addition for debugging, I don't need to keep it.